lib/phoenix_live_view/tag_engine.ex

```diff
@@ -1160,6 +1163,93 @@ defmodule Phoenix.LiveView.TagEngine do
     end
   end
 
+  defp handle_keyed_comprehension(key_expr, state, tag_meta) do
+    # for now, we only support plain tags because we don't know if a function component
+    # renders to a single tag which is required by live components
+    case state.tag_handler.classify_type(tag_meta.tag_name) do
```

Comment:
I am not a big fan that we are calling `classify_type`. We already knew the type before we called `pop_special_attributes`, so I recommend either passing the type in or passing a boolean saying if `:key` is allowed or not.

For the error handling, I don't think we need to say what you have. We could just say: "keyed-comprehensions are only supported on regular tags, not for #{tag_meta.tag_name}`".